apps/web/src/pages/liquid-staking/[contract].tsx

```diff
@@ -0,0 +1,170 @@
+import { useTranslation } from '@pancakeswap/localization'
+import { CardBody, Text, RowBetween, Box, Flex, Image } from '@pancakeswap/uikit'
+import { AppBody, AppHeader } from 'components/App'
+import CurrencyInputPanel from 'components/CurrencyInputPanel'
+import Page from 'views/Page'
+import { useCurrency } from 'hooks/Tokens'
+import { LightGreyCard } from 'components/Card'
+import { useRouter } from 'next/router'
+import { useEffect, useMemo, useState } from 'react'
+import { getFullDisplayBalance } from '@pancakeswap/utils/formatBalance'
+import BigNumber from 'bignumber.js'
+import { ChainId } from '@pancakeswap/sdk'
+import { useCurrencyBalance } from 'state/wallet/hooks'
+import { BIG_ZERO } from '@pancakeswap/utils/bigNumber'
+import AddToWalletButton from 'components/AddToWallet/AddToWalletButton'
+import { maxAmountSpend } from 'utils/maxAmountSpend'
+import { LiquidStakingFAQs } from 'views/LiquidStaking/components/FAQs'
+import useAccountActiveChain from 'hooks/useAccountActiveChain'
+import { LiquidStakingList } from 'views/LiquidStaking/constants/types'
+import { fetchLiquidStaking } from 'views/LiquidStaking/hooks/useLiquidStakingList'
+import StakeInfo from 'views/LiquidStaking/components/StakeInfo'
+import { useExchangeRate } from 'views/LiquidStaking/hooks/useExchangeRate'
+import LiquidStakingButton from 'views/LiquidStaking/components/LiquidStakingButton'
+
+const LiquidStakingStakePage = () => {
+  const { t } = useTranslation()
+  const router = useRouter()
+  const { account, chainId } = useAccountActiveChain()
+  const [stakeAmount, setStakeAmount] = useState('')
+  const [showPage, setShowPage] = useState(false)
+  const [selectedList, setSelectedList] = useState<null | LiquidStakingList>(null)
+
+  useEffect(() => {
+    const contract: string = (router?.query?.contract as string) ?? ''
```

Comment:
let use `useParams` instead of query. The query of next js router combines url params with search params. Quite annoying if they somehow override each other

https://nextjs.org/docs/app/api-reference/functions/use-params

Reply:
Let's keep this untouched and will have further discussion about the app router